x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -108,7 +108,7 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	regionsList, err := getRegions(svcEC2)
 	if err != nil {
 		err = errors.Wrap(err, "getRegions failed")
-		logp.Error(err)
+		logp.NewLogger(aws.ModuleName).Errorf(err.Error())
```

Comment:
you can keep this logger instance in the metricset object to avoid creating new instances every time a new log is stored, I see you do the same later on in several places

Reply:
@exekias Thanks for the suggestion! https://github.com/elastic/beats/pull/10142/files#diff-019d1f0f86b061ba4cc82e0fa0c1e30aR50 is this what you are looking for?

Reply:
yes, thank you for updating!, btw `ec2` part here is implicit, we normally call these just `logger`

Reply:
Changed to `logger` 👍  Thank you!